Formula/newt.rb

```diff
@@ -22,14 +22,12 @@ class Newt < Formula
 
   depends_on "gettext"
   depends_on "popt"
+  depends_on "python@3.9"
   depends_on "s-lang"
 
-  on_linux do
-    depends_on "python@3.10"
-  end
-
   def install
-    args = ["--prefix=#{prefix}", "--without-tcl"]
+    with_python = "python#{Language::Python.major_minor_version(Formula["python@3.9"])}"
+    args = %W[--prefix=#{prefix} --without-tcl -with-python=#{with_python}]
```

Comment:
I was able to use `python@3.10` with a few tweaks
```suggestion
    args = %W[--prefix=#{prefix} --without-tcl --with-python=#{with_python}]
```

Reply:
[`configure` considers single-dash and double-dash options equivalent](https://bitbucket.org/berkeleylab/upcxx/issues/345/configure-with-single-dash-arguments). This behavior is not well known and arguably can be surprising. If you feel strongly about this, please feel free to make this change.

Reply:
I've made this change.